### PR TITLE
Moving log and fmt PrintX to be logrus Info and Debug

### PR DESCRIPTION
**Pull request by paganotoni**
This is related with #765

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -63,25 +64,25 @@ func (a *App) Serve() error {
 	go func() {
 		// gracefully shut down the application when the context is cancelled
 		<-ctx.Done()
-		fmt.Println("Shutting down application")
+		logrus.Info("Shutting down application")
 
 		err := a.Stop(ctx.Err())
 		if err != nil {
-			fmt.Println(err)
+			logrus.Debug(err)
```
Comment (markbates) [inline app.go]: This should probably be sent to the Error logger.

File: app.go [hunk 2]
```diff
@@ -63,25 +64,25 @@ func (a *App) Serve() error {
...
 		}
 
 		if !a.WorkerOff {
 			// stop the workers
-			fmt.Println("Shutting down worker")
+			logrus.Info("Shutting down worker")
 			err = a.Worker.Stop()
 			if err != nil {
-				fmt.Println(err)
+				logrus.Debug(err)
```
Comment (markbates) [inline app.go]: This should probably be sent to the Error logger.

File: app.go [hunk 3]
```diff
@@ -63,25 +64,25 @@ func (a *App) Serve() error {
...
 			}
 		}
 
 		err = server.Shutdown(ctx)
 		if err != nil {
-			fmt.Println(err)
+			logrus.Debug(err)
```
Comment (markbates) [inline app.go]: This should probably be sent to the Error logger.

File: app.go [hunk 4]
```diff
@@ -125,7 +126,7 @@ func (a *App) Serve() error {
 func (a *App) Stop(err error) error {
 	a.cancel()
 	if err != nil && errors.Cause(err) != context.Canceled {
-		fmt.Println(err)
+		logrus.Debug(err)
```
Comment (markbates) [inline app.go]: This should probably be sent to the Error logger.

File: buffalo/cmd/build/target.go [hunk 5]
```diff
@@ -4,7 +4,7 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/sirupsen/logrus"
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/build/target.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/build/transform.go [hunk 6]
```diff
@@ -6,8 +6,8 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/build/transform.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/build/transform_main.go [hunk 7]
```diff
@@ -6,9 +6,9 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/build/transform_main.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/destroy/resource.go [hunk 8]
```diff
@@ -9,6 +9,7 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/destroy/resource.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/dev.go [hunk 9]
```diff
@@ -2,11 +2,11 @@ package cmd
 
 import (
 	"context"
-	"fmt"
 	"os"
 	"os/exec"
 	"runtime"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/dev.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/new.go [hunk 10]
```diff
@@ -1,11 +1,12 @@
 package cmd
 
 import (
-	"fmt"
 	"os"
 	"os/user"
 	"path/filepath"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/new.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/plugins.go [hunk 11]
```diff
@@ -2,11 +2,11 @@ package cmd
 
 import (
 	"fmt"
-	"log"
 	"os"
 	"os/exec"
 	"sync"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/plugins.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/plugins.go [hunk 12]
```diff
@@ -23,7 +23,7 @@ func plugs() plugins.List {
 		_plugs, err = plugins.Available()
 		if err != nil {
 			_plugs = plugins.List{}
-			log.Printf("error loading plugins %s\n", err)
+			logrus.Info("error loading plugins %s\n", err)
```
Comment (markbates) [inline buffalo/cmd/plugins.go]: Error logger

File: buffalo/cmd/root.go [hunk 13]
```diff
@@ -2,9 +2,9 @@ package cmd
 
 import (
 	"errors"
-	"fmt"
 	"os"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/root.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/root.go [hunk 14]
```diff
@@ -41,7 +41,7 @@ var RootCmd = &cobra.Command{
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
 	if err := RootCmd.Execute(); err != nil {
-		fmt.Printf("Error: %s\n\n", err)
+		logrus.Debugf("Error: %s\n\n", err)
```
Comment (markbates) [inline buffalo/cmd/root.go]: Error logger

File: buffalo/cmd/setup.go [hunk 15]
```diff
@@ -3,13 +3,13 @@ package cmd
...
 	"bytes"
 	"context"
-	"fmt"
 	"os"
 	"os/exec"
 	"strings"
 
 	"golang.org/x/sync/errgroup"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/setup.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/test.go [hunk 16]
```diff
@@ -2,14 +2,14 @@ package cmd
...
 	"bytes"
-	"fmt"
 	"io"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"regexp"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/test.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: buffalo/cmd/version.go [hunk 17]
```diff
@@ -1,8 +1,7 @@
 package cmd
 
 import (
-	"fmt"
-
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline buffalo/cmd/version.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: generators/action/action.go [hunk 18]
```diff
@@ -6,6 +6,7 @@ import (
 	"path/filepath"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline generators/action/action.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: generators/assets/webpack/webpack.go [hunk 19]
```diff
@@ -1,10 +1,10 @@
 package webpack
 
 import (
-	"fmt"
 	"os/exec"
 	"path/filepath"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline generators/assets/webpack/webpack.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: grifts.go [hunk 20]
```diff
@@ -35,7 +36,7 @@ func secretGrift() {
 		rx := regexp.MustCompile(`(\W+)`)
 		bb = rx.ReplaceAll(bb, []byte(""))
 		s := randx.String(6) + string(bb)
-		fmt.Println(s[:127])
+		logrus.Infof(s[:127])
```
Comment (markbates) [inline grifts.go]: This should be just a plain println so people can pipe the output easily into something else.

File: grifts.go [hunk 21]
```diff
@@ -49,9 +50,8 @@ func middlewareGrift(a *App) {
 }
 
 func printMiddleware(a *App) {
-	fmt.Println(fmt.Sprintf("-> %s", a.Name))
-	fmt.Println(a.Middleware.String())
-	fmt.Println()
+	logrus.Infof(fmt.Sprintf("-> %s", a.Name))
```
Comment (markbates) [inline grifts.go]: I don't think you need the Sprintf here

File: grifts/shoulders.go [hunk 22]
```diff
@@ -1,14 +1,14 @@
...
 import (
-	"fmt"
 	"html/template"
 	"os"
 	"os/exec"
 	"path"
 	"sort"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline grifts/shoulders.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: logger.go [hunk 23]
```diff
@@ -1,8 +1,8 @@
 package buffalo
 
 import (
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline logger.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: logger_formatter.go [hunk 24]
```diff
@@ -1,6 +1,6 @@
 package buffalo
 
-// I really don't want to have this, but until (if) https://github.com/sirupsen/logrus/pull/606 is merged we're stuck with all this code. And yes, this is ALL needed just to remove some blank space in the logs
+// I really don't want to have this, but until (if) https://github.com/Sirupsen/logrus/pull/606 is merged we're stuck with all this code. And yes, this is ALL needed just to remove some blank space in the logs
```
Comment (markbates) [inline logger_formatter.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: logger_formatter.go [hunk 25]
```diff
@@ -12,7 +12,7 @@ import (
 	"sync"
 	"time"
 
-	"github.com/sirupsen/logrus"
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline logger_formatter.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: options.go [hunk 26]
```diff
@@ -3,10 +3,11 @@ package buffalo
 import (
 	"context"
 	"fmt"
-	"log"
 	"net/http"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline options.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: plugins/plugins.go [hunk 27]
```diff
@@ -83,12 +83,12 @@ func askBin(path string) Commands {
 	cmd.Stderr = bb
 	err := cmd.Run()
 	if err != nil {
-		fmt.Printf("[PLUGIN] error loading plugin %s: %s\n%s\n", path, err, bb.String())
+		logrus.Infof("[PLUGIN] error loading plugin %s: %s\n%s\n", path, err, bb.String())
```
Comment (markbates) [inline plugins/plugins.go]: Error logger

File: plugins/plugins.go [hunk 28]
```diff
@@ -83,12 +83,12 @@ func askBin(path string) Commands {
...
 		return commands
 	}
 	err = json.NewDecoder(bb).Decode(&commands)
 	if err != nil {
-		fmt.Printf("[PLUGIN] error loading plugin %s: %s\n", path, err)
+		logrus.Infof("[PLUGIN] error loading plugin %s: %s\n", path, err)
```
Comment (markbates) [inline plugins/plugins.go]: error logger

File: render/template.go [hunk 29]
```diff
@@ -3,12 +3,12 @@ package render
 import (
 	"html/template"
 	"io"
-	"log"
 	"os"
 	"path/filepath"
 	"sort"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline render/template.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: request_logger.go [hunk 30]
```diff
@@ -3,10 +3,10 @@ package buffalo
 import (
 	"time"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline request_logger.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

File: worker/simple.go [hunk 31]
```diff
@@ -5,8 +5,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/Sirupsen/logrus"
```
Comment (markbates) [inline worker/simple.go]: The repo is now lower case `sirupsen`. Can you please undo these import changes?

**Comment by paganotoni** [comment]
@markbates thanks for the feedback, all these have been implemented, let me know how is it looking when you have the chance, thanks again!

File: buffalo/cmd/build.go [hunk 32]
```diff
@@ -57,7 +56,7 @@ var xbuildCmd = &cobra.Command{
 			return errors.WithStack(err)
 		}
 
-		fmt.Printf("\nYou application was successfully built at %s\n", filepath.Join(b.Root, b.Bin))
+		logrus.Info("\nYou application was successfully built at %s\n", filepath.Join(b.Root, b.Bin))
```
Comment (markbates) [inline buffalo/cmd/build.go]: this needs to be `Infof`

File: buffalo/cmd/build/apkg.go [hunk 33]
```diff
@@ -91,7 +91,7 @@ func (b *Builder) buildADatabase() error {
 			logrus.Debugf("no sqlite usage in database.yml detected, applying the nosqlite tag")
 			b.Tags = append(b.Tags, "nosqlite")
 		} else if !b.Static {
-			fmt.Println("you are building a SQLite application, please consider using the `--static` flag to compile a static binary")
+			logrus.Debugf("you are building a SQLite application, please consider using the `--static` flag to compile a static binary")
```
Comment (markbates) [inline buffalo/cmd/build/apkg.go]: This can be just `Debug`

File: buffalo/cmd/build/apkg.go [hunk 34]
```diff
@@ -91,7 +91,7 @@ func (b *Builder) buildADatabase() error {
...
 		}
 	} else {
 		logrus.Debugf("no database.yml detected, applying the nosqlite tag")
```
Comment (markbates) [inline buffalo/cmd/build/apkg.go]: This can be just `Debug`

File: buffalo/cmd/destroy/resource.go [hunk 35]
```diff
@@ -51,22 +52,22 @@ func confirm(msg string) bool {
 func removeTemplates(fileName string) {
 	if YesToAll || confirm("Want to remove templates? (Y/n)") {
 		templatesFolder := fmt.Sprintf(filepath.Join("templates", fileName))
-		fmt.Printf("- Deleted %v folder\n", templatesFolder)
+		logrus.Info("- Deleted %v folder\n", templatesFolder)
```
Comment (markbates) [inline buffalo/cmd/destroy/resource.go]: This needs to be `Infof`

File: buffalo/cmd/destroy/resource.go [hunk 36]
```diff
@@ -51,22 +52,22 @@ func confirm(msg string) bool {
...
 		os.Remove(filepath.Join("actions", fmt.Sprintf("%v.go", fileName)))
 
-		fmt.Printf("- Deleted %v\n", fmt.Sprintf("actions/%v_test.go", fileName))
+		logrus.Infof("- Deleted %v\n", fmt.Sprintf("actions/%v_test.go", fileName))
 		os.Remove(filepath.Join("actions", fmt.Sprintf("%v_test.go", fileName)))
 
 		content, err := ioutil.ReadFile(filepath.Join("actions", "app.go"))
 		if err != nil {
-			fmt.Println("[WARNING] error reading app.go content")
+			logrus.Info("[WARNING] error reading app.go content")
```
Comment (markbates) [inline buffalo/cmd/destroy/resource.go]: This should be `Warn` and we can remove the `warning` label

File: buffalo/cmd/destroy/resource.go [hunk 37]
```diff
@@ -75,11 +76,11 @@ func removeActions(fileName string) {
 
 		err = ioutil.WriteFile(filepath.Join("actions", "app.go"), []byte(newContents), 0)
 		if err != nil {
-			fmt.Println("[WARNING] error writing new app.go content")
+			logrus.Info("[WARNING] error writing new app.go content")
```
Comment (markbates) [inline buffalo/cmd/destroy/resource.go]: This should be `Error` and it should problem return the error.

File: buffalo/cmd/dev.go [hunk 38]
```diff
@@ -39,7 +39,7 @@ This behavior can be changed in your .buffalo.dev.yml file.`,
 					cause = err.Error()
 				}
 			}
-			fmt.Printf(msg, cause)
+			logrus.Error(msg, cause)
```
Comment (markbates) [inline buffalo/cmd/dev.go]: This should be `Errorf`

File: buffalo/cmd/setup.go [hunk 39]
```diff
@@ -214,7 +214,7 @@ func nodeCheck() error {
 }
 
 func run(cmd *exec.Cmd) error {
-	fmt.Printf("--> %s\n", strings.Join(cmd.Args, " "))
+	logrus.Info("--> %s\n", strings.Join(cmd.Args, " "))
```
Comment (markbates) [inline buffalo/cmd/setup.go]: This should be `Infof`

File: generators/action/action.go [hunk 40]
```diff
@@ -100,7 +101,7 @@ func (act Generator) findActionsToAdd(path string) []meta.Name {
 	for _, action := range act.Actions {
 		funcSignature := fmt.Sprintf("func %s%s(c buffalo.Context) error", act.Name.Camel(), action.Camel())
 		if strings.Contains(string(fileContents), funcSignature) {
-			fmt.Printf("--> [warning] skipping %v%v since it already exists\n", act.Name.Camel(), action.Camel())
+			logrus.Infof("--> [warning] skipping %v%v since it already exists\n", act.Name.Camel(), action.Camel())
```
Comment (markbates) [inline generators/action/action.go]: This should be `Warnf` and we can remove the `warning` label

File: generators/action/action.go [hunk 41]
```diff
@@ -121,7 +122,7 @@ func (act Generator) findHandlersToAdd(path string) []meta.Name {
 	for _, action := range act.Actions {
 		funcSignature := fmt.Sprintf("app.GET(\"/%s/%s\", %s%s)", act.Name.URL(), action.URL(), act.Name.Camel(), action.Camel())
 		if strings.Contains(string(fileContents), funcSignature) {
-			fmt.Printf("--> [warning] skipping %s from app.go since it already exists\n", funcSignature)
+			logrus.Infof("--> [warning] skipping %s from app.go since it already exists\n", funcSignature)
```
Comment (markbates) [inline generators/action/action.go]: This should be `Warnf` and we can remove the `warning` label

File: generators/action/action.go [hunk 42]
```diff
@@ -142,7 +143,7 @@ func (act Generator) findTestsToAdd(path string) []meta.Name {
 	for _, action := range act.Actions {
 		funcSignature := fmt.Sprintf("func (as *ActionSuite) Test_%v_%v() {", act.Name.Camel(), action.Camel())
 		if strings.Contains(string(fileContents), funcSignature) {
-			fmt.Printf("--> [warning] skipping Test_%v_%v since it already exists\n", act.Name.Camel(), action.Camel())
+			logrus.Infof("--> [warning] skipping Test_%v_%v since it already exists\n", act.Name.Camel(), action.Camel())
```
Comment (markbates) [inline generators/action/action.go]: This should be `Warnf` and we can remove the `warning` label

File: generators/assets/webpack/webpack.go [hunk 43]
```diff
@@ -26,7 +26,7 @@ func (w Generator) Run(root string, data makr.Data) error {
 
 	// if there's no npm, return!
 	if _, err := exec.LookPath("npm"); err != nil {
-		fmt.Println("Could not find npm. Skipping webpack generation.")
+		logrus.Infof("Could not find npm. Skipping webpack generation.")
```
Comment (markbates) [inline generators/assets/webpack/webpack.go]: This should be `Info`

File: grifts.go [hunk 44]
```diff
@@ -35,7 +36,7 @@ func secretGrift() {
 		rx := regexp.MustCompile(`(\W+)`)
 		bb = rx.ReplaceAll(bb, []byte(""))
 		s := randx.String(6) + string(bb)
-		fmt.Println(s[:127])
+		println(s[:127])
```
Comment (markbates) [inline grifts.go]: This should still be using `fmt.Println`

File: options.go [hunk 45]
```diff
@@ -121,7 +122,7 @@ func optionsWithDefaults(opts Options) Options {
 		secret := envy.Get("SESSION_SECRET", "")
 		// In production a SESSION_SECRET must be set!
 		if opts.Env == "production" && secret == "" {
-			log.Println("WARNING! Unless you set SESSION_SECRET env variable, your session storage is not protected!")
+			logrus.Warn("WARNING! Unless you set SESSION_SECRET env variable, your session storage is not protected!")
```
Comment (markbates) [inline options.go]: we can remove the `warning` label

**Comment by paganotoni** [comment]
@markbates next round of changes applied :)

File: grifts.go [hunk 46]
```diff
@@ -49,9 +50,8 @@ func middlewareGrift(a *App) {
 }
 
 func printMiddleware(a *App) {
-	fmt.Println(fmt.Sprintf("-> %s", a.Name))
-	fmt.Println(a.Middleware.String())
-	fmt.Println()
+	logrus.Infof("-> %s", a.Name)
```
Comment (markbates) [inline grifts.go]: this function needs to go back to using `Println`.

**Comment by paganotoni** [comment]
@markbates all set :)

File: grifts.go [hunk 47]
```diff
@@ -49,9 +49,8 @@ func middlewareGrift(a *App) {
 }
 
 func printMiddleware(a *App) {
-	fmt.Println(fmt.Sprintf("-> %s", a.Name))
-	fmt.Println(a.Middleware.String())
-	fmt.Println()
+	fmt.Printf("-> %s ", a.Name)
```
Comment (markbates) [inline grifts.go]: Sorry for this, but this statement is missing a newline. Last change, I swear! :)

**Comment by paganotoni** [comment]
lol, now all set

**Comment by markbates** [comment]
Awesome! Thanks.

**Comment by paganotoni** [comment]
Thank you for all the help here!

**Pull request merged**